src/main/java/org/testng/internal/TestListenerHelper.java

```diff
@@ -19,31 +20,60 @@ private TestListenerHelper() {
     // Utility class. Defeat instantiation.
   }
 
-  static void runPreConfigurationListeners(ITestResult tr, List<IConfigurationListener> listeners) {
+  static void runPreConfigurationListeners(ITestResult tr, ITestNGMethod tm, List<IConfigurationListener> listeners) {
     for (IConfigurationListener icl : listeners) {
       icl.beforeConfiguration(tr);
+      try {
+        icl.beforeConfiguration(tr, tm);
+      } catch (Exception e) {
+        handleException(e);
+      }
     }
   }
 
   static void runPostConfigurationListeners(
-      ITestResult tr, List<IConfigurationListener> listeners) {
+      ITestResult tr, ITestNGMethod tm, List<IConfigurationListener> listeners) {
     for (IConfigurationListener icl : listeners) {
       switch (tr.getStatus()) {
         case ITestResult.SKIP:
           icl.onConfigurationSkip(tr);
+          try {
+            icl.onConfigurationSkip(tr, tm);
+          } catch (Exception e) {
+            handleException(e);
+          }
           break;
         case ITestResult.FAILURE:
           icl.onConfigurationFailure(tr);
+          try {
+            icl.onConfigurationFailure(tr, tm);
+          } catch (Exception e) {
+            handleException(e);
+          }
           break;
         case ITestResult.SUCCESS:
           icl.onConfigurationSuccess(tr);
+          try {
+          icl.onConfigurationSuccess(tr,tm);
+          } catch (Exception e) {
+            handleException(e);
+          }
           break;
         default:
           throw new AssertionError("Unexpected value: " + tr.getStatus());
       }
     }
   }
 
+  //This method is added because Gradle which builds TestNG seems to be using an older version
+  //of TestNG that doesn't know about the new methods that we added and so it causes
+  //the TestNG build to keep failing.
+  private static void handleException(Exception e) {
```

Comment:
rename to something like: `ignoreInternalGradleException`

Reply:
sure will do that

Reply:
Fixed.